src/mscorlib/src/System/TimeZoneInfo.cs

```diff
@@ -1005,8 +982,42 @@ private AdjustmentRule GetPreviousAdjustmentRule(AdjustmentRule rule)
             }
             return cachedData.m_readOnlySystemTimeZones;
         }
-#endif // FEATURE_WIN32_REGISTRY
 
+        private static void PopulateAllSystemTimeZones(CachedData cachedData)
+        {
+#if FEATURE_WIN32_REGISTRY
+            PermissionSet permSet = new PermissionSet(PermissionState.None);
+            permSet.AddPermission(new RegistryPermission(RegistryPermissionAccess.Read, c_timeZonesRegistryHivePermissionList));
+            permSet.Assert();
+            
+            using (RegistryKey reg = Registry.LocalMachine.OpenSubKey(
+                c_timeZonesRegistryHive,
+#if FEATURE_MACL
+                RegistryKeyPermissionCheck.Default,
+                System.Security.AccessControl.RegistryRights.ReadKey
+#else
+                false
+#endif
+                )) {
+
+                if (reg != null) {
+                    foreach (string keyName in reg.GetSubKeyNames()) {
+                        TimeZoneInfo value;
+                        Exception ex;
+                        TryGetTimeZone(keyName, false, out value, out ex, cachedData);  // populate the cache
+                    }
+                }
+            }
+#else // FEATURE_WIN32_REGISTRY
```

Comment:
Should this be:
```C#
#elif PLATFORM_UNIX
```
to keep it consistent with TryGetTimeZoneByFile being defined uned PLATFORM_UNIX?